browser-fetching/index.md

```diff
@@ -0,0 +1,80 @@
+* **Partially updating the page** in response to user interactivity
+  _For example, fetching and displaying the current stock price for a user-provided stock index_
+* **Submitting forms** and displaying a success or error messages without needing to reload an entire page.
+  _For example, Adding an item to a shopping cart_
+
+Fetch's interface only requires a single argument—a `url` address to send the request to. The second argument is an object for specifying various options.
+
+```javascript
+  fetch(url, options) // return a Promise
+```
+As with most asynchronous APIs in JavaScript, `fetch` returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) that either resolves (succeeds) or rejects (fails) after receiving a response.
+
+Below is a minimal example of using fetch to request a random dog image from [dog.ceo](https://dog.ceo). The image returned is then displayed in the page along with the full JSON response. 
+
+If you'd like to see the HTTP requests sent during fetches in your browser, open the Web Developer Tools and select the Network tab.
+
+
+```javascript
+const fetchDog = () => fetch('https://dog.ceo/api/breeds/image/random')
+  // Resolve the JSON returned from the url
+  .then(response => response.json())
+  // Display the image and the raw JSON in the page
+  .then(result => {
+    console.log(result)
+    document.querySelector('#dog').src = result.message
+    document.querySelector('#dog-results').innerHTML = JSON.stringify(result, null, 2)
+  })
+  // Log any errors to the developer console
+  .catch(error => console.error(error))
+```
+
+<script>
+const fetchDog = () => fetch('https://dog.ceo/api/breeds/image/random')
+  .then(response => response.json())
+  .then(result => {
+    console.log(result)
+    document.querySelector('#dog').src = result.message
+    document.querySelector('#dog-results').innerHTML = JSON.stringify(result, null, 2)
+  })
+  .catch(error => console.error(error))
+</script>
+<button onclick="fetchDog()">Call the `fetchDog` function</button>
+
+#### JSON from the response:
+<pre class="language-javascript" id="dog-results">Nothing fetched yet.</pre>
+
+#### Image from the response:
+<div style="width: 400px;">
+  <img id="dog" style="width: 100%;" alt="Random dog image" src="https://via.placeholder.com/400" />
+</div>
+
+Below is more complex example that uses multiple options to `POST` a request to a weather API:
+
+```
+TODO
+```
+
+## Summary
+
+The previous examples only hint at the capabilities of the Fetch API. Read the ["Using the Fetch API"](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) page on the Mozilla Developer Network (MDN) for more comprehensive examples and detail on the Fetch API. Fetch isn't limited to web browsers—it's also available in [Node.js v18](https://nodejs.org/en/blog/announcements/v18-release-announce/#fetch-experimental) and can be used on the server.
```

Comment:
I think we want to emphasize more the magic of APIs. This would be a good place for a little more persuasive writing. We're always transparent, but in this case we try to persuade them a little more that this is something worth learning.